tests/model_forms/test_modelchoicefield.py

```diff
@@ -373,3 +373,28 @@ class CategoriesForm(forms.Form):
         )
         with self.assertNumQueries(2):
             template.render(Context({'form': CategoriesForm()}))
+
+    def test_custom_invalid_choice_message(self):
+        f = forms.ModelChoiceField(
+            queryset=Category.objects.all(),
+            error_messages={
+                'invalid_choice': 'Select a valid choice. "%(value)s" is not one of the available choices.'
+            }
+        )
+        with self.assertRaises(ValidationError) as ctx:
+            f.clean("-1234")
+        self.assertEqual(
+            ctx.exception.messages,
+            ['Select a valid choice. "-1234" is not one of the available choices.']
+        )
+
+    def test_custom_invalid_choice_message_without_value(self):
+        f = forms.ModelChoiceField(
+            queryset=Category.objects.all(),
+            error_messages={
+                'invalid_choice': 'Select a valid choice'
+            }
+        )
+        with self.assertRaises(ValidationError) as ctx:
+            f.clean("-1234")
+        self.assertEqual(ctx.exception.messages, ['Select a valid choice'])
```

Comment:
This is already covered by `forms_tests.tests.test_error_messages.ModelChoiceFieldErrorMessagesTestCase.test_modelchoicefield`. We can remove it.